File: app/src/main/java/at/shockbytes/dante/network/amazon/AmazonItemLookupApi.java
package at.shockbytes.dante.network.amazon;

import at.shockbytes.dante.util.books.BookSuggestion;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 13.02.2017.
 */

public interface AmazonItemLookupApi {

    String SERVICE_ENDPOINT = "http://webservices.amazon.com/onca/";

    @GET("json")
    Observable<BookSuggestion> downloadBookSuggestion(@Query("Service") String service,
                                                      @Query("Operation") String operation,
                                                      @Query("ResponseGroup") String respGroup,
                                                      @Query("SearchIndex") String searchIndex,
                                                      @Query("IdType") String idType,
                                                      @Query("ItemId") String isbn,
                                                      @Query("AWSAccessKeyId") String accessKey,
                                                      @Query("AssociateTag") String associateTag,
                                                      @Query("Timestamp") String timestamp,
                                                      @Query("Signature") String requestSignature);

}

File: app/src/main/java/at/shockbytes/dante/network/amazon/AmazonItemLookupApi.kt
package at.shockbytes.dante.network.amazon

import at.shockbytes.dante.book.BookSuggestion
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Author: Martin Macheiner
 * Date: 13.02.2017
 */
interface AmazonItemLookupApi {

    @GET("json")
    fun downloadBookSuggestion(
        @Query("Service") service: String,
        @Query("Operation") operation: String,
        @Query("ResponseGroup") respGroup: String,
        @Query("SearchIndex") searchIndex: String,
        @Query("IdType") idType: String,
        @Query("ItemId") isbn: String,
        @Query("AWSAccessKeyId") accessKey: String,
        @Query("AssociateTag") associateTag: String,
        @Query("Timestamp") timestamp: String,
        @Query("Signature") requestSignature: String
    ): Observable<BookSuggestion>

    companion object {
        const val SERVICE_ENDPOINT = "http://webservices.amazon.com/onca/"
    }
}
